### ISSUE TITLE
 Segmentation fault when running lammps with deepmd + kspace + bond/angle information

### ISSUE THREAD

**Issue by njzjz** [deepmodeling/deepmd-kit#4250]
Hi,

I'm trying to run lammps with hybrid/scaled pair_style to combine deepmd model and empirical model. Therefore, I'm using atom_style full with data file that contains coordinates, bonds, and angles. To use lj/cut/coul/long I'm also trying to use kspace (kspace_style pppm 1e-5 kspace_modify gewald 0.45). However, I'm getting segmentation fault. After some tests, I identified that I can run successfully deepmd model with full atom style and data file that includes bond/angle information but without kspace. Or I can run deepmd with kspace and full atom style but bonds and angles are removed from the data file. But when I combine all of them I'm getting segmentation fault. Is there some reason for this? And can it be fixed? 

Any help will be greatly appreciated.

Thanks,
Dina

Here is an error message:
![error_message](https://github.com/user-attachments/assets/ab99a77c-d7b1-4d7f-bba7-ec641a823ec7)
I also tried to run with different lammps/deepmd, but getting similar error:
![error_message2](https://github.com/user-attachments/assets/e5680ae8-bbcb-4bbb-ba48-900192917bab)
Input lammps file:
![input_file](https://github.com/user-attachments/assets/eb189647-13bb-45f5-9afe-01ac80e2598b)
Data file for lammps:
![data_file](https://github.com/user-attachments/assets/27793f7d-be58-49dc-930e-1a8f6da1310a)

_Originally posted by @dkussainova in https://github.com/deepmodeling/deepmd-kit/discussions/4184_

**Comment by HasnainMSajid** [comment]
I'm also facing the same issue.
My system consists of a MOF with guest molecules and I want to apply deep potentials for MOF atoms and LJ/coul for guest atoms and guest-mof interactions. I'm using atoms_style full because of coulombic interactions. In the data file, I need to specify the bonds and angles of CO2 (otherwise, it breaks apart during energy minimization), but I get segmentation fault. Any guidance on this problem will be appreciated. 

Lammps input and data files are attached. 


[in.MD](https://github.com/user-attachments/files/17527590/in.MD)
[data.MD](https://github.com/user-attachments/files/17527584/data.MD)

**Comment by njzjz** [comment]
> I'm also facing the same issue.
> My system consists of a MOF with guest molecules and I want to apply deep potentials for MOF atoms and LJ/coul for guest atoms and guest-mof interactions. I'm using atoms_style full because of coulombic interactions. In the data file, I need to specify the bonds and angles of CO2 (otherwise, it breaks apart during energy minimization), but I get segmentation fault. Any guidance on this problem will be appreciated. 
> 
> Lammps input and data files are attached. 
> 
> 
> [in.MD](https://github.com/user-attachments/files/17527590/in.MD)
> [data.MD](https://github.com/user-attachments/files/17527584/data.MD)
> 
>  

@HasnainMSajid Could you provide `graph-compress.pb` in `in.MD`?

**Comment by HasnainMSajid** [comment]
The file can be accessed here.
https://1drv.ms/u/s!AgFT-vPyGbY_gxbhLF9qUcZpCtGA?e=eRh8Xq

**Comment by njzjz** [comment]
```
Thread 1 "lmp" received signal SIGSEGV, Segmentation fault.
0x00007ffff1f97770 in deepmd::NeighborListData::shuffle (this=0x41dbef0, 
    fwd_map=std::vector of length 7714, capacity 7714 = {...})
    at /workspaces/deepmd-kit/source/api_cc/src/common.cc:262
262             jlist[ii][jj] = fwd_map[jlist[ii][jj]];
(gdb) print ii
$1 = 193
(gdb) print jj
$2 = 58
(gdb) print jlist.size()
$3 = 1596
(gdb) print jlist[ii].size()
$4 = 238
(gdb) print jlist[ii][jj]
$5 = -2147477131
(gdb) print nloc
$6 = 7714
```

**Comment by njzjz** [comment]
The negative neighbor list is not related to DeePMD-kit. I checked the neighbor list at the top of the `compute` method, and still got the negative neighbor list passed in:

```cpp
void PairDeepMD::compute(int eflag, int vflag) {
  for (int ii = 0; ii < list->inum; ++ii) {
    for (int jj = 0; jj < list->numneigh[ii]; ++jj) {
      if (list->firstneigh[ii][jj] < 0) {
        printf("list.firstneigh[%d][%d] = %d \n", ii, jj, list->firstneigh[ii][jj]);
        throw;
      }
    }
  }
```

Got:

> list.firstneigh[193][58] = -2147477131 

@HasnainMSajid You could consider reporting the neighbor list bug to LAMMPS.

**Comment by njzjz** [comment]
Fixed by #4269.
